web/web.go

```diff
@@ -0,0 +1,694 @@
+		http.Error(w, fmt.Sprintf("Connection Error: %v. Check Configuration again.", err), http.StatusInternalServerError)
+		return
+	}
+	app.sourceDB = sourceDB
+	app.dbName = config.Database
+	app.driver = config.Driver
+	w.WriteHeader(http.StatusOK)
+}
+
+func convertSchemaSQL(w http.ResponseWriter, r *http.Request) {
+	if app.sourceDB == nil || app.dbName == "" || app.driver == "" {
+		http.Error(w, fmt.Sprintf("Database is not configured or Database connection is lost. Please set configuration and connect to database."), http.StatusNotFound)
+		return
+	}
+	conv := internal.MakeConv()
+	var err error
+	switch app.driver {
+	case "mysql":
+		err = mysql.ProcessInfoSchema(conv, app.sourceDB, app.dbName)
+	case "postgres":
+		err = postgres.ProcessInfoSchema(conv, app.sourceDB)
+	default:
+		http.Error(w, fmt.Sprintf("Driver : '%s' is not supported", app.driver), http.StatusBadRequest)
+		return
+	}
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Schema Conversion Error : %v", err), http.StatusNotFound)
+		return
+	}
+	app.conv = nil
+	app.conv = conv
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(conv)
+}
+
+// Dump config is used for dump file approach.
+type DumpConfig struct {
+	Driver   string `json:"Driver"`
+	FilePath string `json:"Path"`
+}
+
+func convertSchemaDump(w http.ResponseWriter, r *http.Request) {
+	reqBody, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Body Read Error : %v", err), http.StatusInternalServerError)
+		return
+	}
+	var dc DumpConfig
+	err = json.Unmarshal(reqBody, &dc)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Request Body parse error : %v", err), http.StatusBadRequest)
+		return
+	}
+	f, err := os.Open(dc.FilePath)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("failed to open the test data file: %v", err), http.StatusNotFound)
+		return
+	}
+	conv, err := conversion.SchemaConv(dc.Driver, &conversion.IOStreams{In: f, Out: os.Stdout}, 0)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Schema Conversion Error : %v", err), http.StatusNotFound)
+		return
+	}
+	app.conv = nil
+	app.conv = conv
+	app.driver = dc.Driver
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(conv)
+}
+
+// TODO: Add Index key statements.
+func getDDL(w http.ResponseWriter, r *http.Request) {
+	c := ddl.Config{Comments: true, ProtectIds: false}
+	var tables []string
+	for t := range app.conv.SpSchema {
+		tables = append(tables, t)
+	}
+	sort.Strings(tables)
+	ddl := make(map[string]string)
+	for _, t := range tables {
+		ddl[t] = app.conv.SpSchema[t].PrintCreateTable(c)
+	}
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(ddl)
+}
+
+func getSummary(w http.ResponseWriter, r *http.Request) {
+	reports := internal.AnalyzeTables(app.conv, nil)
+	summary := make(map[string]string)
+	for _, t := range reports {
+		var body string
+		for _, x := range t.Body {
+			body = body + x.Heading + "\n"
+			for i, l := range x.Lines {
+				body = body + fmt.Sprintf("%d) %s.\n", i+1, l) + "\n"
```

Comment:
Just curious about the \n added by the Sprintf and then another concatenated?
Do you need both?
If so, maybe just \n\n in the Sprintf?

Reply:
yeah to have one line separation both are needed, moved second \n inside as mentioned